File: app/frontend/trade-app/src/app/shared/models/trade-response.ts
import {InstrumentResponse} from './instrument-response';
import {AccountResponse} from './account-response';
import {Unit} from '../enums/unit';
import {Status} from '../enums/status';
import {DeliveryType} from '../enums/delivery-type';
import {Direction} from 'node:tty';

export interface TradeResponse {
  id:number,
  instrument: InstrumentResponse,
  account: AccountResponse,
  quantity:number,
  price:number,
  direction:Direction,
  deliveryType:DeliveryType,
  unit:Unit,
  status:Status,
}

File: app/backend/trade-app/src/main/java/com/trade_app/dto/response/TradeResponse.java
package com.trade_app.dto.response;

import com.trade_app.domain.enums.DeliveryType;
import com.trade_app.domain.enums.Direction;
import com.trade_app.domain.enums.Status;
import com.trade_app.domain.enums.Unit;
import com.trade_app.dto.request.AccountRequest;
import com.trade_app.dto.request.InstrumentRequest;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class TradeResponse {
    private Long id;
    private InstrumentResponse instrument;
    private AccountResponse account;
    private Integer quantity;
    private Float price;
    private Direction direction;
    private DeliveryType deliveryType;
    private Unit unit;
    private Status status;
}
